hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -137,4 +126,27 @@
       throw new HoodieIOException("Unable to read metadata for instant " + hoodieInstantOption.get(), io);
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Get pending instant.
+   *
+   * @param metaClient
+   * @return
+   */
+  public static Set<String> getInflightInstants(HoodieTableMetaClient metaClient) {
+    // collect pending deltaCommit/commit/compaction/clustering
+    return metaClient
+            .getActiveTimeline()
+            .filterInflights().getInstants().map(HoodieInstant::getTimestamp).collect(Collectors.toSet());
```

Comment:
Do we also need to include requested instants here?

Reply:
My apologies, I missed submitting the comment before.

Reply:
@xiarixiaoyao what's your take on this?

Reply:
i agree with suryaprasanna， we should include requested instants
If it is an ordinary commit（General write operation）, this place does not need to be considered since we will deal it  later, but if it is a commit of competition or clustering, this place must be considered.

let‘s me fixed it, and add UT for it